### PR TITLE
Enable on demand sync for RepositoryVersionDistribution

### PR DESCRIPTION
**Pull request by fao89**
https://pulp.plan.io/issues/7568
closes #7568

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/content/handler.py [hunk 1]
```diff
@@ -442,7 +442,12 @@ async def _match_and_stream(self, path, request):
             except ObjectDoesNotExist:
                 pass
             else:
-                return self._serve_content_artifact(ca, headers)
+                if ca.artifact:
+                    return self._serve_content_artifact(ca, headers)
+                else:
+                    return await self._stream_content_artifact(
```
Comment (ipanova) [inline pulpcore/content/handler.py:448]: :+1:

**Comment by mdellweg** [comment]
Nice.
Do we have tests in that area? I doubt that it can easily be tested with pulp_file.

File: CHANGES/7568.bugfix [hunk 2]
```diff
@@ -0,0 +1 @@
+Enable on demand sync for RepositoryVersionDistribution
```
Comment (ipanova) [inline CHANGES/7568.bugfix:1]: how about "enable content streaming.."

**Comment by ipanova** [comment]
> Nice.
> Do we have tests in that area? I doubt that it can easily be tested with pulp_file.

I don't think this can be tested with the pulp_file since it has a publication, it should be a plugin that has distribution only, like pulp_container just not subclassed handler. I think pulp_ansible uses directly pulpcore's content handler

**Comment by fao89** [comment]
> Nice.
> Do we have tests in that area? I doubt that it can easily be tested with pulp_file.

it is being tested here: https://github.com/pulp/pulp_ansible/pull/373

**Review by ipanova — APPROVED** [review]

**Review by bmbouter — APPROVED** [review]
This looks good to me. Thanks @fao89 !

**Pull request merged**